daemon/mgr/image.go

```diff
@@ -174,6 +177,25 @@ func (mgr *ImageManager) PullImage(ctx context.Context, ref string, authConfig *
 	return mgr.StoreImageReference(ctx, img)
 }
 
+// PushImage push image to specified registry.
+func (mgr *ImageManager) PushImage(ctx context.Context, name, tag string, authConfig *types.AuthConfig, out io.Writer) error {
+	ref, err := reference.Parse(name)
+	if err != nil {
+		return err
+	}
+
+	if tag == "" {
+		ref = reference.WithDefaultTagIfMissing(ref)
+	} else {
+		ref = reference.WithTag(ref, tag)
+	}
+
+	stream := jsonstream.New(out, nil)
+	err = mgr.client.PushImage(ctx, ref.String(), authConfig, stream)
+	stream.Wait()
```

Comment:
I don't think we should wait here. If the image is not found, the `stream.Wait()` will be dead end. WDYT?

Reply:
I think it ok here, since in stream.New(), it have do the add, but move it to `ctrd/image.go` make more sense.
```
// New creates a 'JSONStream' instance and use a goroutine to recv/send data.
func New(out io.Writer, f Formater) *JSONStream {
    format := f
    if format == nil {
        format = newDefaultFormat()
    }   

    stream := &JSONStream{
        w:     out,
        cache: make(chan interface{}, 16),
        f:     format,
    }   

    stream.wg.Add(1)
```